src/_pytest/assertion/rewrite.py

```diff
@@ -291,12 +294,12 @@ def _read_pyc(source, pyc, trace=lambda x: None):
     Return rewritten code if successful or None if not.
     """
     try:
-        fp = open(pyc, "rb")
+        fp = open(fspath(pyc), "rb")
```

Comment:
this one shouldn't need `fspath` should it?

Reply:
if we need hax for py35 to use pathlib I'd prefer we don't use pathlib until we drop python3.5

Reply:
I don't think it is hack to be honest... it is a compatibility function, like we have many others in `compat`. It is literally a if/else with an implementation specific to py35.

This request would need me to review the entire patch, and I really would like to avoid that unless we have good reason.

Reply:
I agree that this particular case is just a compat function, but pathlib in general feels like a liability if it's going to blow up mysteriously in python3.5

Reply:
> if it's going to blow up mysteriously in python3.5

Definitely agree for external APIs, but this is an internal change. And it doesn't blow mysteriously: remove that call and it does fail on python3.5 immediately, with a good error message. Or perhaps you mean something else by mysteriously?

Reply:
this particular usage doesn't, but I'm not confident of test coverage about all of our uses of such functions

there's also potentially more surface area than expected here since it's an import hook -- but maybe it's fine

I just have my reservations about pathlib being a good idea when there's some pretty easy footguns :) (and I don't like pathlib myself)

Reply:
Well, let's hold this PR until we drop py35 drops then. 👍

Reply:
I'm fine with this PR btw, just _concerned_ is all ;) -- I think if you use `exist_ok=True` below I'll press approve :D